compat/mingw.c

```diff
@@ -383,6 +383,21 @@ static wchar_t *normalize_ntpath(wchar_t *wbuf)
 	return wbuf;
 }
 
+static int do_unlink(const char *pathname, const wchar_t *wpathname,
+		     int (*unlinker)(const wchar_t *))
+{
+	char temppath[MAX_PATH];
+	wchar_t wtemppath[MAX_LONG_PATH];
+	strncpy(temppath, pathname, sizeof(temppath) - 8);
+	strcat(temppath, ".XXXXXX");
+	mktemp(temppath);
```

Comment:
We could use _wmktemp directly so we don't need to convert back and forth between char and wide char.

Reply:
I'm not sure. mingw_mktemp has some magic that wraps _mktemp[1]. Can we safely skip it?

[1]
```
	/* we need to return the path, thus no long paths here! */
	if (xutftowcs_path(wtemplate, template) < 0)
		return NULL;

	if (is_dir_sep(template[0]) && !is_dir_sep(template[1]) &&
	    iswalpha(wtemplate[0]) && wtemplate[1] == L':') {
		/* We have an absolute path missing the drive prefix */
		offset = 2;
	}
```